idl-parser/src/ffi/ast/mod.rs

```diff
@@ -1,30 +1,127 @@
 use crate::ast;
-use std::{slice, str};
+use std::{
+    ffi::{c_char, CString},
+    slice, str,
+};
 
 pub mod visitor;
 
+#[repr(C)]
+pub struct ParseResult {
+    program: *mut Program,
```

Comment:
Btw, why do we need a pointer here? Isn't allocating just `ParseResult` enough?

Reply:
It points to Program, which is not repr(C). So having it in a separate pointer makes sense